### ISSUE TITLE
Fails to open

### ISSUE THREAD

**Issue by hasen6** [caprica/vlcj-javafx-demo#41]
I downloaded the demo and run it but it throws this error on JDK 12 and JavaFX 16 on Mac (other Javafx programs run no problem):
```
Exception in Application constructor
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:464)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:363)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:1051)
Caused by: java.lang.RuntimeException: Unable to construct Application instance: class uk.co.caprica.vlcj.javafx.demo.VlcjJavaFxApplication
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:890)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
        at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$8(LauncherImpl.java:802)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:474)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:447)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:389)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:446)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
Caused by: java.lang.NoClassDefFoundError: uk/co/caprica/vlcj/binding/RuntimeUtil
        at uk.co.caprica.vlcj@4.7.0/uk.co.caprica.vlcj.factory.discovery.strategy.LinuxNativeDiscoveryStrategy.supported(LinuxNativeDiscoveryStrategy.java:47)
        at uk.co.caprica.vlcj@4.7.0/uk.co.caprica.vlcj.factory.discovery.NativeDiscovery.discover(NativeDiscovery.java:111)
        at uk.co.caprica.vlcj@4.7.0/uk.co.caprica.vlcj.factory.MediaPlayerFactory.discoverNativeLibrary(MediaPlayerFactory.java:176)
        at uk.co.caprica.vlcj@4.7.0/uk.co.caprica.vlcj.factory.MediaPlayerFactory.<init>(MediaPlayerFactory.java:108)
        at uk.co.caprica.vlcj@4.7.0/uk.co.caprica.vlcj.factory.MediaPlayerFactory.<init>(MediaPlayerFactory.java:163)
        at uk.co.caprica.vlcj.javafx.demo/uk.co.caprica.vlcj.javafx.demo.VlcjJavaFxApplication.<init>(VlcjJavaFxApplication.java:30)
        ... 11 more
Caused by: java.lang.ClassNotFoundException: uk.co.caprica.vlcj.binding.RuntimeUtil
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 17 more
Exception running application uk.co.caprica.vlcj.javafx.demo.VlcjJavaFxApplication
```

**Comment by caprica** [comment]
How are you running it?
Did you set up the classpath properly?
If you don't use Maven or an IDE you will probably have to manually setup some things.

**Comment by hasen6** [comment]
I'm running it in VSCode with Maven. I have managed to run other Javafx programs like a simple one I made with MediaPlayer before I found that didn't have all functionality I need while VLCJ does.

It doesn't show any errors in the code in the IDE and it's correctly showing version 16 of JavaFX. Not sure what needs to be set up, I thought it would just run since it's a demo.

**Comment by caprica** [comment]
Well, the demo does in fact "just run" for me when I pull it into a proper IDE.

Getting it to work in VSCode is another matter.

As for the error you posted:

```
Caused by: java.lang.ClassNotFoundException: uk.co.caprica.vlcj.binding.RuntimeUtil
```

That error has an umabiguous meaning. It says that you have not got the classpath setup properly, it can't find a vlcj library class that it depends on.

Why is that the case? I don't know, I do use VSCode but I do not use it for Java development. Something is wrong with your VSCode project setup or run configuration or whatever.

**Comment by hasen6** [comment]
Ok thanks for explaining what the error means. I guess maybe it's set up for your IDE or something. I wanted to try this demo since using VLCJ in another project I made myself which is working, but when the video opens it's a video window but the screen is black despite the fact the sound of the video can be heard.

**Comment by caprica** [comment]
If you want an easier life, download IntelliJ Community Edition (it's free), and import this project into it. Then run the supplied launcher class. It will work.

I can't say why this project is not working for you in VSCode, but that's a local problem that I can't help you with. This project works just fine ordinarily.

If audio is working in your JavaFX application but you're seeing a black screen instead of video then I would advise you to carefully look at how you constructed your Stage and how the components are added to it - are you sure the video component (the ImageView associated with the ImageViewVideoSurface) was added to your stage correctly?

**Comment by hasen6** [comment]
Someone had the same issue here it seems https://github.com/caprica/vlcj/issues/887 and you said "https://github.com/caprica/vlcj/issues/887". Since I'm also on Mac perhaps it just doesn't work for Mac?

I have IntelliJ sitting on my hard drive so I imported the demo as you said and selected Maven but I get `Cannot resolve uk.co.caprica:vlcj:4.5.0-SNAPSHOT`. Not really familiar with IntelliJ so not sure what that is.

**Comment by hasen6** [comment]
This is the code for my entire project. I even added a frame from Swing trying to get it to show properly. All I'm trying to do is get a basic video window to open with video displayed properly:

```
package com.french;

import javafx.application.Application;
import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.geometry.Insets;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.image.ImageView;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.HBox;
import javafx.scene.paint.Color;
import javafx.stage.Stage;
import uk.co.caprica.vlcj.factory.MediaPlayerFactory;
import uk.co.caprica.vlcj.player.base.MediaPlayer;
import uk.co.caprica.vlcj.player.base.MediaPlayerEventAdapter;
import uk.co.caprica.vlcj.player.component.EmbeddedMediaPlayerComponent;
import uk.co.caprica.vlcj.player.embedded.EmbeddedMediaPlayer;

import java.util.List;

import javax.swing.JFrame;

import static uk.co.caprica.vlcj.javafx.videosurface.ImageViewVideoSurfaceFactory.videoSurfaceForImageView;

public class App extends Application {
    private final MediaPlayerFactory mediaPlayerFactory;
    private final EmbeddedMediaPlayer embeddedMediaPlayer;
    private final EmbeddedMediaPlayerComponent mediaPlayerComponent = new EmbeddedMediaPlayerComponent();
    private ImageView videoImageView;
    private String path = "/Volumes/Movies/stuff/projects/JavaFX/learnfrench/src/media/video.mp4";
    private Button bntPlay;
    private HBox hbox = new HBox(10);

    public App() {
        this.mediaPlayerFactory = new MediaPlayerFactory();
        this.embeddedMediaPlayer = mediaPlayerFactory.mediaPlayers().newEmbeddedMediaPlayer();
        this.embeddedMediaPlayer.events().addMediaPlayerEventListener(new MediaPlayerEventAdapter() {
            @Override
            public void playing(MediaPlayer mediaPlayer) {
            }

            @Override
            public void paused(MediaPlayer mediaPlayer) {
            }

            @Override
            public void stopped(MediaPlayer mediaPlayer) {
            }

            @Override
            public void timeChanged(MediaPlayer mediaPlayer, long newTime) {
            }
        });
    }

    @Override
    public void init() {
        this.videoImageView = new ImageView();
        this.videoImageView.setPreserveRatio(true);

        embeddedMediaPlayer.videoSurface().set(videoSurfaceForImageView(this.videoImageView));
    }

    @Override
    public final void start(Stage primaryStage) throws Exception {
        List<String> params = getParameters().getRaw();
        // params.add(new File("/Volumes/Movies/stuff/projects/JavaFX/french/src/media/peppa.mp4").getAbsolutePath());
        // System.out.println(params.size());
        // if (params.size() != 1) {
        //     System.out.println("Specify a single MRL");
        //     System.exit(-1);
        // }

        BorderPane root = new BorderPane();
        root.setStyle("-fx-background-color: black;");

        // videoImageView.fitWidthProperty().bind(root.widthProperty());
        // videoImageView.fitHeightProperty().bind(root.heightProperty());

        root.widthProperty().addListener((observableValue, oldValue, newValue) -> {
            // If you need to know about resizes
        });

        root.heightProperty().addListener((observableValue, oldValue, newValue) -> {
            // If you need to know about resizes
        });

        bntPlay = new Button(">");
        hbox.getChildren().add(bntPlay);
        hbox.setPadding(new Insets(10));
        bntPlay.setOnAction(new EventHandler<ActionEvent>() {

            @Override
            public void handle(ActionEvent event) {
                mediaPlayerComponent.mediaPlayer().media().play(path);
            }

        });
        root.setCenter(videoImageView);
        root.setBottom(hbox);
        Scene scene = new Scene(root, 1200, 675, Color.BLACK);
        primaryStage.setTitle("vlcj JavaFX");
        primaryStage.setScene(scene);
        primaryStage.show();

        // embeddedMediaPlayer.media().play(params.get(0));
        // embeddedMediaPlayer.media().play(path);
        // mediaPlayerComponent.mediaPlayer().media().play(path);

        embeddedMediaPlayer.controls().setPosition(0.4f);

        JFrame frame = new JFrame("vlcj Tutorial");

        frame.setContentPane(mediaPlayerComponent);

        frame.setLocation(100, 100);
        frame.setSize(1050, 600);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setVisible(true);
        mediaPlayerComponent.mediaPlayer().media().play(path);
        // embeddedMediaPlayer.media().play(path);
    }

    @Override
    public final void stop() {
        embeddedMediaPlayer.controls().stop();
        embeddedMediaPlayer.release();
        mediaPlayerFactory.release();
    }

    public static void main(String[] args) {
        launch();
    }
}
```

**Comment by caprica** [comment]
Regarding the snapshot dependency, that's unfortunate that I left it that way in the project, you can change it to the latest non-snapshot version of vlcj, i.e. 4.7.1 rather than 4.5.0-SNAPSHOT.

**Comment by hasen6** [comment]
Still get errors in IntelliJ with 4.7.1 and vlcj-javafx 1.0.2.

/Volumes/Movies/stuff/projects/JavaFX/vlcj-javafx-demo-master/src/test/java/uk/co/caprica/vlcj/javafx/test/JavaFXDirectRenderingTest.java
Error:(39, 26) java: cannot find symbol
  symbol:   class PixelBuffer
  location: package javafx.scene.image
Error:(188, 13) java: cannot find symbol
  symbol:   class PixelBuffer
  location: class uk.co.caprica.vlcj.javafx.test.JavaFXDirectRenderingTest
Error:(388, 31) java: cannot find symbol
  symbol:   class PixelBuffer
  location: class uk.co.caprica.vlcj.javafx.test.JavaFXDirectRenderingTest.JavaFxBufferFormatCallback
/Volumes/Movies/stuff/projects/JavaFX/vlcj-javafx-demo-master/src/test/java/uk/co/caprica/vlcj/javafx/test/WritableImageDemo.java
Error:(12, 26) java: cannot find symbol
  symbol:   class PixelBuffer
  location: package javafx.scene.image
Error:(69, 13) java: cannot find symbol
  symbol:   class PixelBuffer
  location: class uk.co.caprica.vlcj.javafx.test.WritableImageDemo
Error:(139, 36) java: cannot find symbol
  symbol:   class PixelBuffer
  location: class uk.co.caprica.vlcj.javafx.test.WritableImageDemo.FXBufferFormatCallback

**Comment by caprica** [comment]
In IntelliJ right-click the pom.xml and choose "add as maven project" or some other option similar to that.

PixelBuffer is a JavaFX class, which IIRC came with JavaFX 13, so for some reason your project can't see JavaFX.

**Comment by hasen6** [comment]
Yeah this is why I don't use IntelliJ, I tried it before and it's just one problem after the other so I stick to VSCode for everything. There wasn't any option like you say, there was only a 'Maven' sub folder when right clicking on pom.xml and in there there is 'reimport' and 'unlink maven projects' which didn't change anything.

![屏幕快照 2021-07-15 上午12 25 16](https://user-images.githubusercontent.com/15882674/125639026-f3900a02-1555-4ccf-80a0-57dc0291bd19.png)

My IntelliJ is version 2019.3 since it cannot update itself. It just updates, restarts and then is still the same version as before it tried to update. I think I remember before I had to create an actual JavaFX project with IntelliJ to get it to accept it, a project from elsewhere would never just work.

So you didn't find anything In the code I posted above where I got audio with a black screen in VSCode? I really think perhaps it may be as you said in the other thread, it just doesn't work on Mac with any kind of recent version of Java.

**Comment by caprica** [comment]
I know of no reason why 2019.3 woudn't work for this, but I see no reason not to install the latest version.

You said you expect it to just work... https://youtu.be/yzmokfajN0o, but I do use up-to-date tools.

IntelliJ is so much better than VSCode for Java it's not even in the same ballpark, and I say this as someone that really likes VSCode (at least for React/Typescript work).

**Comment by caprica** [comment]
Does this...
```
root.setCenter(videoImageView);
```
..._definitely_ expand the ImageView to fit?

Otherwise this is hard to understand what's not working for you, if you have audio playback working it can only be some sort of layout problem.

You can also try "-vvv" when you create the MediaPlayerFactory and see if there's anything of note in the native log messages that will be produced.

**Comment by caprica** [comment]
When you mention running on Mac, this sort of "callback" or "direct" rendering absolutely _should_ work on Mac - it's the heavyweight native media player that no longer works due to no more AWT in the JVM on macOS.

**Comment by hasen6** [comment]
> I know of no reason why 2019.3 woudn't work for this, but I see no reason not to install the latest version.

Well like I said above, IntelliJ fails to update every single time. It downloads the update, restarts and when it opens again it's still the same version it was. Every time I want to update I have to download the whole thing from their website, delete the old version and install the new one. The problem doesn't go away with a new version either. That's the reason.

VSCode doesn't have this problem at all.

> Does this...
> 
> ```
> root.setCenter(videoImageView);
> ```
> 
> ..._definitely_ expand the ImageView to fit?

Well I've no idea. I get a video window which is big enough to show a video if that's what you mean, the only problem is it's black. Seems to me that that would just put the video in the centre, not expand it. I tried different videos and it's the same with them. The code I gave above is the whole code I'm using. I'm assuming it works for you on I'm guessing, on pc? If it works for you then you'd have to imagine it's a Mac Java thing.

>When you mention running on Mac, this sort of "callback" or "direct" rendering absolutely should work on Mac - it's the heavyweight native media player that no longer works due to no more AWT in the JVM on macOS.

Well I don't know what's going on then.

**Comment by hasen6** [comment]
I manually updated my IntelliJ to the latest version then followed your video to the letter and I got:

/Volumes/Movies/stuff/projects/JavaFX/vlcj-javafx-demo/src/test/java/uk/co/caprica/vlcj/javafx/test/JavaFXDirectRenderingTest.java:
Error:(39, 26) java: cannot find symbol
  symbol:   class PixelBuffer
  location: package javafx.scene.image

**Comment by caprica** [comment]
PixelBuffer came with JavaFX 13.

**Comment by hasen6** [comment]
> PixelBuffer came with JavaFX 13.

We're talking about the JDK version, not the JavaFX version are we not? I have JavaFX 16, however the JDK version in run configuration is 12.

![屏幕快照 2021-07-15 下午3 04 23](https://user-images.githubusercontent.com/15882674/125731674-c9b841fb-4721-416e-9ee0-7ddcbd59c2fa.png)

**Comment by caprica** [comment]
```
/Volumes/Movies/stuff/projects/JavaFX/vlcj-javafx-demo/src/test/java/uk/co/caprica/vlcj/javafx/test/JavaFXDirectRenderingTest.java:
Error:(39, 26) java: cannot find symbol
  symbol:   class PixelBuffer
  location: package javafx.scene.image
```

Cannot find symbol PixelBuffer.

**Comment by caprica** [comment]
Also, when I change the run configuration to use JDK 11 it still works.

**Comment by hasen6** [comment]
Well that was the only thing different following your tutorial.

**Comment by caprica** [comment]
The only question worth addressing at this point in time is why your setup can not locate PixelBuffer.

I suppose there's a chance the JavaFX PixelBuffer is not supported on macOS, but I would be very surprised if that were the case - it's part of the public API for JavaFX13+.

Inside IntelliJ, or the equivalent in VSCode, press CTRL+N (or the equivalent shortcut on macOS for Navigate Class)  and type "PixelBuffer" without quotes, ensuring that the top right of the popup says "All Places", and see if PixelBuffer can be found.

**Comment by caprica** [comment]
I am actually pretty sure I had this working on my iMac, but it would have been quite a while ago.

**Comment by hasen6** [comment]
![屏幕快照 2021-07-15 下午3 59 22](https://user-images.githubusercontent.com/15882674/125736560-cf2beadc-26a4-4f6f-8ff9-549d72cbf1c9.png)

**Comment by caprica** [comment]
Interesting, yet the error you posted from your IDE clearly states this class can not be found.

**Comment by hasen6** [comment]
> Interesting, yet the error you posted from your IDE clearly states this class can not be found.

Well you're talking about a piece of software that asks if you want to update, downloads the update, restarts and opens having not been updated whatsoever. I think at least it's a shadow of what it is on the pc. This is why I thought it would be simpler to get it to work with VSCode.

I think somehow it's using a different version of JavaFX than it looks like it's using.

**Comment by caprica** [comment]
By all means you should switch back to VSCode if you're comfortable there, but it's then on you to get it working, I don't have any other suggestions.

I don't experience the same problems you do, for me things do just work.

**Comment by hasen6** [comment]
> By all means you should switch back to VSCode if you're comfortable there, but it's then on you to get it working, I don't have any other suggestions.
> 
> I don't experience the same problems you do, for me things do just work.

Yeah I think you're right it's a Mac issue, for both IntelliJ and VLCJ. Both my code from vscode AND this demo code in IntelliJ just work for you while they just _don't_ work for me, even with the same exact code I posted above, or with me following your actions to the letter with IntelliJ and the demo project.

I don't see how it can be anything other than that it just doesn't work on Mac...

**Comment by hasen6** [comment]
Oh but incidentally my code from above that opens to a black screen in VSCode also works in IntelliJ but also with a black screen and audio only so I don't think it's a VSCode issue.

**Comment by caprica** [comment]
It does work on Mac, I have an iMac that I occasionally test on. It's just been a while since I've done it so I don't recall the details. I might try it again later if I find some time.

**Comment by hasen6** [comment]
> It does work on Mac, I have an iMac that I occasionally test on. It's just been a while since I've done it so I don't recall the details. I might try it again later if I find some time.

Yeah if you can get my code above running on Mac, that would suggest an OS version issue.

**Comment by caprica** [comment]
OK, not a word of a lie.

My iMac IntelliJ version was 2019.x, I don't recall specifically which version but 2019 something. I did the automatic update and it updated it without issue to the current 2021 version.

I then did exactly the same thing on my iMac as I showed you in the earlier video on my main Linux workstation - from inside IntelliJ, clone the project from GitHub and run it.

And it just worked.

I don't know what else to tell you.

**Comment by hasen6** [comment]
> I don't know what else to tell you.

Well the OS version you run would be useful.

I did a system print of the version of JavaFX in the demo project when IntelliJ runs and it's 14.9 so really should be no problem. It's inexplicable why it can't find pixel buffer.

But since the code I posted above actually does open for me, with no errors it would be more interesting to see if you could run that on Mac without seeing a black screen like I do.

**Comment by caprica** [comment]
I run the latest version of the OS, always. And I really can't see how a different OS version is relevant here.

I helped you as much as I can. "Works for me" doesn't help you too much, but that's the truth and this is likely as far as I can take it.

**Comment by hasen6** [comment]
> I run the latest version of the OS, always. And I really can't see how a different OS version is relevant here.

Considering neither of us can even begin to explain this problem I think anything is possible. It can find pixel buffer using navigate class, it **IS** using JavaFX 14 at runtime as confirmed by runtime system print but yet it can't find pixel buffer when running.

The IntelliJ update problem is pretty common and the threads I read there is no solution either.

>I helped you as much as I can. "Works for me" doesn't help you too much, but that's the truth and this is likely as far as I can take it.

Yeah with the demo. I guess I should post a separate issue on the main VLCJ regarding the black screen thing rather than discuss it here since as you say it's not an issue with the demo. I think it's likely that's an entirely different unrelated issue.

**Comment by hasen6** [comment]
Out of interest, what version of the JDK were you running on your Mac?

**Comment by caprica** [comment]
It was JDK 14. That just happened to be what was current the last time I used Java on that computer I suppose.

**Comment by hasen6** [comment]
Ok I'm using JDK 12 so that's always a possibility for the errors for the demo not launching or the black screen on my code.

**Comment by caprica** [comment]
Having looked at your code, honestly it's got some problems.

You are mixing Swing and JavaFX, not advised, but OK.

But on the media player you are adding the media player component to the JFrame, but using a JavaFX ImageView as your video surface. That simply can't work.

For JavaFX you should be using CallbackMediaPlayer.

If you do want to use Swing in your JavaFX application, then you can only use something like a Canvas for your media player component, not the image view stuff.

**Comment by hasen6** [comment]
Thanks for looking at my code. The reason I started adding Swing code in was because all the solutions to this problem were with Swing, it already wasn't working prior to adding Swing - same black screen issue. Also your documentation examples are all Swing.

Talking of your documentation, you stated this: _"What this means in practice is that the EmbeddedMediaPlayer, and associated component, will not work on any Java version later than 1.6 on macOS - unless you use an alternate video surface like the CallbackVideoSurface that implements "direct rendering". This is covered in a later tutorial."_

Wouldn't this be the problem I'm facing in my code above since I'm not using Java 1.6 and I AM using EmbeddedMediaPlayer?

I'll try updating my OS today to solve the weird pixel buffer problem since I'm sure you'll agree there's no logical reason why it would 'just work' on your mac and just not work on mine.

**Comment by hasen6** [comment]
Got the code to work with audio and video but has a new issue. Posted in separate thread.

**Comment by caprica** [comment]
It's as simple as with Swing the video surface is a Canvas.
With JavaFX the video surface is an ImageView.

So in the JavaFX case you are not supposed to add the media player component to the UI like you have done with `frame.setContentPane(mediaPlayerComponent);`.

Anyway, it seems you are a step forward so let's close this.

**Comment by hasen6** [comment]
I think what happened is it wasn't working a few months back when I tried it so I started adding other stuff from Swing etc but in fact the problem was my JavaFX was not the right version. Then after updating it these last few days, I never reverted the rest of the code back to how it was a couple of months back. It seems that fixed it without me knowing. Still no idea about the weirdness with IntelliJ and the demo was about but all I needed was a basic working project and ideally with VSCode anyway.
